include/rcutils/time.h

```diff
@@ -36,7 +36,10 @@ extern "C"
 #define RCUTILS_US_TO_NS(microseconds) ((microseconds) * 1000LL)
 
 /// Convenience macro to convert nanoseconds to seconds.
-#define RCUTILS_NS_TO_S(nanoseconds) ((nanoseconds) / (1000LL * 1000LL * 1000LL))
+#define RCUTILS_NS_TO_S(nanoseconds) \
+  ((1e-9 * (double)((int32_t)(nanoseconds % 1000000000l))) + \
+  ((double)((int32_t)((nanoseconds - ((int32_t)(nanoseconds % 1000000000l))) / 1000000000l))))
```

Comment:
just curious, is there any accuracy difference from this relying on FPU? i did not check the assembly files though.

```suggestion
#define RCUTILS_NS_TO_S(nanoseconds) ((double)(nanoseconds) / (1e9))
```

the following sample gives me the exact same floating value so far.

```cpp
#include <stdio.h>
#include <stdint.h>

#define NANO_TO_SEC(nanoseconds) ((double)(nanoseconds) / 1e9)

#define RCUTILS_NS_TO_S(nanoseconds)                         \
  ((1e-9 * (double)((int32_t)(nanoseconds % 1000000000l))) + \
  ((double)((int32_t)((nanoseconds - ((int32_t)(nanoseconds % 1000000000l))) / 1000000000l))))

int main() {
  int64_t nanoseconds = 1234567890123456789; // Example nanoseconds
  double seconds = NANO_TO_SEC(nanoseconds);
  printf("Nanoseconds: %ld \nSeconds: %lf\n", nanoseconds, seconds);

  seconds = RCUTILS_NS_TO_S(nanoseconds);
  printf("Nanoseconds: %ld \nSeconds: %lf\n", nanoseconds, seconds);

  return 0;
}
```

Reply:
[This `geometry2` issue](https://github.com/ros2/geometry2/issues/34) should explain the whole reasoning as to why this improves accuracy, but pretty much scaling each set of nanoseconds, prevents excess rounding, in case the user needs additional accuracy. I'm almost positive `printf` will do some rounding for visual help, but technically if you make this test:

```
 EXPECT_EQ(RCUTILS_NS_TO_S(-1999999999ll), -1.999999999);  // int64_t
```
You'll get a result of:

>  Expected equality of these values:
        ((1e-9 * (double)((int32_t)(-1999999999ll % 1000000000l))) + ((double)((int32_t)((-199999999       9ll - ((int32_t)(-1999999999ll % 1000000000l))) / 1000000000l))))
          Which is: -2
        -2

When you should've just [left this](https://github.com/ros2/rcutils/blob/431e7f6cd19fe207247b1619ff12c17401339cfa/test/test_time.cpp#L96)